blog/2015-05-27-so-you-want-to-transform-part-1.adoc

```diff
@@ -0,0 +1,215 @@
+the "Source Type (XML)" page appear next. If we had selected another
+type, we'd see that page here.
+
+In this case, we have an XML Schema document (abc-order.xml) that we
+want to transform from. So we'll click on the Source File field's "..."
+button to browse for our file, select abc-order.xml, and click Next to
+continue. If you have a particularly complex schema, you may need to
+select from the root elements you wish to key on for the transformation.
+
+image:images/transformation_tools/new-fuse-transformation-wizard-source-xml-details.png[image]
+
+== New Fuse Transformation Wizard, Target JSON Details
+
+Since we selected JSON as the Target Type on the Start page, we'll see
+the "Target Type (JSON)" page appear next. If we had selected another
+type, we'd see that page here.
+
+In this case, we have an JSON Schema document (xyz-order-schema.json)
+that we want to transform to. So we'll click on the Target File field's
+"..." button to browse for our file, select xyz-order-schema.json, and
+click Finish to continue.
+
+image:images/transformation_tools/new-fuse-transformation-wizard-target-json-details.png[image]
+
+== Finishing Up...
+
+As soon as the wizard is done, you'll notice two things.
+
+. The Fuse Transformation Editor opens so you can edit your new Dozer
+configuration file.
+
+. The Camel Editor has been updated with the new Transformation endpoint
+you just created. (Notice the "dirty" flag on the "camel-context.xml"
+tab.)
+
+For now, we'll close our new Transformation and go back to the Camel
+route to finish hooking up the new endpoint.
+
+image:images/transformation_tools/finishing-up.png[image]
+
+== Going back to the Camel editor
+
+When we select the Camel editor again, we notice that a new endpoint
+reference has been added to the canvas. We need to now hook that into
+our route so it happens between our source and target steps of this
+simple configuration.
+
+The first thing we need to do is delete the existing link between the
+"src" and "target" elements of the route. You can click on it and hit
+the Delete key on your keyboard or right-click on the arrow and select
+"Remove." When it asks if you want to remove the element, click Yes.
+
+image:images/transformation_tools/going-back-to-the-camel-editor.png[image]
+
+== Wiring up our New Endpoint
+
+Now we can wire up the endpoint.
+
+. If you hover the mouse over the "src" file element, you'll see a
+toolbar pop up. Click and drag the Arrow to our new endpoint.
+
+. Do the same thing from the endpoint to the "target" element.
+
+. Next, save the Camel editor and it will refresh the view to clean up
+your route a little.
+
+image:images/transformation_tools/wiring-up-our-new-endpoint.png[image]
+
+== End for today...
+
+So now we have a new Transformation endpoint that is wired into our
+Camel route.
+
+Next time we'll start doing the actual editing of the Transformation
+itself!
+
+image:images/transformation_tools/end-for-today.png[image]
+
+Thanks for reading!
```

Comment:
we tend to signoff blogs with our name(s) - in case reader forgot who was listed at top ;)